ee/vellum_ee/workflows/display/nodes/base_node_display.py

```diff
@@ -463,38 +463,86 @@ def __init_subclass__(cls, **kwargs: Any) -> None:
 
         register_node_display_class(node_class=node_class, node_display_class=cls)
 
+    def _get_display_class_overrides(self) -> tuple[Optional[str], Optional[str]]:
+        """Extract icon and color from nested Display class if present."""
+        if not hasattr(self._node, "Display"):
+            return None, None
+
+        display_cls = getattr(self._node, "Display")
+        return getattr(display_cls, "icon", None), getattr(display_cls, "color", None)
+
+    def _merge_display_attributes(
+        self,
+        base_kwargs: Dict[str, Any],
+        explicit_icon: Optional[str],
+        explicit_color: Optional[str],
+        display_class_icon: Optional[str],
+        display_class_color: Optional[str],
```

Comment:
[nit] this will scale poorly as we eventually migrate the rest of `BaseNodeDisplay` -> `BaseNode.Display`. 

I believe we can simplify by just fully serializing with `BaseNode.Display`, then overwriting with `BaseNodeDisplay`